### PR TITLE
UI updates to collections search results row layout

### LINKED ISSUE #1745: Collections: Add counts to search results
### Descriptive summary

The count of works and collections should be part of the search results displayed with collections in search results.  This is part of the remaining work from issue #1670 - Calculate the number of items in a collection.

### Rationale

This provides the user additional information and makes collections more visually distinct in search results.

### Expected behavior

The UI for search results needs to be updated to display the work count and subcollection count for collections in search results.  The display should match the mockup (below).

Use the following to get the number of works...
```
<%=collection_presenter.total_viewable_works%>
```
and this for the number of subcollections...
```
<%=collection_presenter.total_viewable_collections%>
```

### Steps to reproduce the behavior

1. Create a collection type that allows discovery
1. Create a collection of that type with at least 3 works and a subcollection (of the same type) as members.
1. Search for the name in the search bar on the home page.

Verify that the collection is in the search results and that the count of works and count of subcollections is correct.

### Mockups

image

### Related work

#1670 Calculate the number of items in a collection
#1739 Collection presenters have counts of viewable members by type and total

### PR DESCRIPTION
**Pull request by elrayle**
Fixes #1745

* handles the UI layout in search results
* calls collection presenter to get counts

### CODE REVIEW

**Comment by adamjarling** [comment]
@elrayle I'm fixing the Hound violations now and will re-push shortly.

File: app/views/catalog/_index_list_default.html.erb [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
+          </div>
+        </div>
+      <% end %>
+    <% end %>
+  </div>
+</div>
+<div class="col-sm-4">
+  <div class="collection-counts-wrapper">
+    <div class="collection-counts-item">
+      <span>2</span>Collections
```
Comment (adamjarling) [inline app/views/catalog/_index_list_default.html.erb]: @elrayle Here's one location where the real count should be wired in.

File: app/views/catalog/_index_list_default.html.erb [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
     </div>
-
-    <div class="clearfix"></div>
+    <div class="collection-counts-item">
+      <span>63</span>Works
```
Comment (adamjarling) [inline app/views/catalog/_index_list_default.html.erb]: @elrayle And here's the second location

File: app/views/catalog/_index_header_list_collection.html.erb [hunk 3]
```diff
@@ -1,3 +1,4 @@
-  <h2>
-    <%= link_to document.title_or_label, [hyrax, document] %>
-  </h2>
+<div class="search-results-title-row">
+  <h3 class="search-result-title"><%= link_to document.title_or_label, [hyrax, document] %></h3>
+  <span class="label label-success">User Collection (wire this up)</span>
```
Comment (elrayle) [inline app/views/catalog/_index_header_list_collection.html.erb]: Where does this show up in the UI?  It looks like `User Collection (wire this up)` either needs to be removed or something needs to be done here.

**Pull request merged**